Technical Review: Image Management

mbologna opened the pull request:
This PR contains a rework of Image Management: I clarified, added/moved/removed some sections.

mbologna commented:
This is ready to review.

Loquacity commented on modules/administration/pages/image-management.adoc:
```diff
@@ -228,29 +232,36 @@ Activation Keys ensure that images using a profile are assigned to the correct c
 
 [[at.images.docker.sourceexample]]
-=== Example Dockerfile and add_packages Script
+==== Example of Dockerfile sources
```
```suggestion
.Example Dockerfile Sources
```

Loquacity commented on modules/administration/pages/image-management.adoc:
```diff
@@ -228,29 +232,36 @@ Activation Keys ensure that images using a profile are assigned to the correct c
-The dockerfile provides access to a specific repository version served by {productname}.
-The following example dockerfile is used by {productname} to trigger a build job on a build host.
+[NOTE]
+.Important instructions in the Dockerfile
```
```suggestion

```

Loquacity commented on modules/administration/pages/image-management.adoc:
```diff
@@ -228,29 +232,36 @@ Activation Keys ensure that images using a profile are assigned to the correct c
+====
+
+[NOTE]
+.Packages Required for Inspecting Your Images
```
```suggestion

```

Loquacity commented on modules/administration/pages/image-management.adoc:
```diff
@@ -228,29 +232,36 @@ Activation Keys ensure that images using a profile are assigned to the correct c
 For example: The [comand]``ARG repo`` parameter and the [command]``echo`` command pointing to the repository file, creates and then injects the correct path into the repository file for the desired channel version.
 
-_The repository version is determined by the activation key that you assigned to your image profile._
+_The repository version is determined by the Activation Key that you assigned to your image profile._
```
```suggestion
The repository version is determined by the Activation Key that you assigned to your image profile.
```

Loquacity commented on modules/administration/pages/image-management.adoc:
```diff
@@ -228,29 +232,36 @@ Activation Keys ensure that images using a profile are assigned to the correct c
+[NOTE]
+.Packages Required for Inspecting Your Images
+====
+To inspect images and provide the package and product list of a container to the {productname} {webui} you must install [package]#python# and [package]#python-xml# within the container.
```
```suggestion
The [package]#python# and [package]#python-xml# packages must be installed in the container.
They are required for inspecting images, and for providing the package and product list of a container to the {productname} {webui}.
```

Loquacity commented on modules/administration/pages/image-management.adoc:
```diff
@@ -228,29 +232,36 @@ Activation Keys ensure that images using a profile are assigned to the correct c
+.Packages Required for Inspecting Your Images
+====
+To inspect images and provide the package and product list of a container to the {productname} {webui} you must install [package]#python# and [package]#python-xml# within the container.
+Without these packages your images will still build, but the package and product list will be unavailable from the {webui}.
```
```suggestion
If you do not install them, images will still build but the package and product list will not available in the {webui}.
```

Loquacity replied:
It seems odd to me that we say "you must install these things" followed immediately by "but it'll work if you don't".

Loquacity commented on modules/administration/pages/image-management.adoc:
```diff
@@ -373,19 +364,17 @@ The status icon for the imported image can also be seen on the ``Overview`` tab
 
-OS images are built by the Kiwi image system.
-They can be of various types: PXE, QCOW2, LiveCD images, and others.
+OS Images are built by the Kiwi image system. The output image is customizable and can be: PXE, QCOW2, LiveCD images, and others.
```
```suggestion
OS Images are built by the Kiwi image system. 
The output image is customizable and can be PXE, QCOW2, LiveCD, or other types of images.
```

Loquacity commented on modules/administration/pages/image-management.adoc:
```diff
@@ -394,42 +383,38 @@ For more information about the Kiwi build system, see the https://doc.opensuse.o
-It is currently not supported to build {sle} 15 images.
-// FIXME CHECKIT
-Building {sle} 15 images requires a {sle} 15 based build host.
+Starting from {sls}{nbsp}15, kiwi-ng will be used instead of the legacy Kiwi.
```
```suggestion
From {sls}{nbsp}15, ``kiwi-ng`` is used instead of the legacy Kiwi.
```

Loquacity commented on modules/administration/pages/image-management.adoc:
```diff
@@ -394,42 +383,38 @@ For more information about the Kiwi build system, see the https://doc.opensuse.o
 
-// FIXME: Provide better reference
-Example scripts are provided in <<at.images.kiwi.sourceexample>>.
+An example of a complete Kiwi repo served by Git can be found at https://github.com/SUSE/manager-build-profiles/tree/master/OSImage
```
```suggestion
For an example of a complete Kiwi repository served by git, see https://github.com/SUSE/manager-build-profiles/tree/master/OSImage
```

Loquacity commented on modules/administration/pages/image-management.adoc:
```diff
@@ -493,16 +471,16 @@ Make sure your build source Kiwi configuration contains `rhn-org-trusted-ssl-cer
+Activation keys are mandatory for OS Image building.
 
 [NOTE]
 .Relationship Between Activation Keys and Image Profiles
```
```suggestion

```

mbologna commented on modules/administration/pages/image-management.adoc:
```diff
@@ -45,22 +50,22 @@ For additional information on setting up Portus as a registry provider, see the
-To build images with {productname}, you will need to create and configure a build host.
-Container build hosts are Salt clients running {sle} 12 or later.
-This section guides you through the initial configuration for a build host.
+To build images with {productname}, you will need to create and configure a Build Host.
```
It's capitalized in the product too:

![image](https://user-images.githubusercontent.com/530992/65505583-d4b5fc80-dec9-11e9-8cf7-7b665ad26291.png)

Loquacity replied:
I would only initcap it in docs if we were directly referring to a UI element, in that case.
So, we would write "Add a container build host by checking the [guimenu]``Container Build Host`` checkbox". Having initcaps scattered through the documentation just makes it harder to read. This article explains it well: https://www.mity.com.au/blog/writing-readable-content-and-why-all-caps-is-so-hard-to-read

mbologna replied:
Fixed

mbologna commented on modules/administration/pages/image-management.adoc:
```diff
@@ -76,15 +81,15 @@ salt '$your_client' state.highstate
 .Relationship Between Activation Keys and Image Profiles
 ====
-To build containers, you will need an activation key that is associated with a channel other than `SUSE Manager Default`.
+To build a container, you will need an Activation Key that is associated with a channel other than `SUSE Manager Default`.
```
I think we have all styles in writing "Activation Keys" in the docs:

```
mbologna@xray:~/Development/doc-susemanager (master =) % rg -r -o "Activation Key" | wc -l  
64
mbologna@xray:~/Development/doc-susemanager (master =) % rg -r -o "Activation key" | wc -l
6
mbologna@xray:~/Development/doc-susemanager (master =) % rg -r -o "activation key" | wc -l
142
```
We should stick to one. Which one would you suggest?

Loquacity replied:
In a heading:
``Create an Activation Key``

At the beginning of a sentence:
``Activation keys are created with ....``

In the middle of a sentence:
``Create an activation key with ... ``

All three are correct :)

mbologna replied:
Fixed

mbologna commented on modules/administration/pages/image-management.adoc:
```diff
@@ -157,56 +161,56 @@ Only lowercase characters are permitted in container labels.
 
-If your git repository is private and not publicly accessible, you need to modify the profile's git URL to include authentication.
-Use this URL format to authenticate with a Github token:
+If your Git repository is private and not publicly accessible, you need to modify the profile's Git URL to include authentication.
```
OK, fixing

mbologna commented on modules/administration/pages/image-management.adoc:
```diff
@@ -6,13 +6,18 @@
+3. Build the image
+4. Push the image in the image store
 
 {productname} supports two distinct build types:
```
Why not `Dockerfile`?

Loquacity replied:
"Docker" is a proper noun, but "dockerfile" is not.

mbologna commented:
OK, I addressed and fixed everything that has been marked as 'resolved'. There are still a couple of unresolved questions, can you please clarify? Thanks for all the input!

mbologna commented on modules/administration/pages/image-management.adoc:
```diff
@@ -76,15 +81,15 @@ salt '$your_client' state.highstate
 
 
-Create an activation key associated with the channel that your images will use.
+The containers built using {productname} will use channel(s) associated to the Activation Key as repositories when building the image. This section will guide you into creating an ad-hoc Activation Key for this purpose.
```
See below comment on Activation Key style

mbologna replied:
Fixed

mbologna commented:
All issues have been resolved. Ready to ship? :shipit:
